powerline/segments/vim/plugin/ctrlp.py

```diff
@@ -0,0 +1,102 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+try:
+	import vim
+except ImportError:
+	vim = object()
+
+from powerline.bindings.vim import create_ruby_dpowerline
+
+
+def initialize():
+	global initialized
+	if initialized:
+		return
+	initialized = True
+	vim.command('echom 3210')
+	vim.command('let g:ctrlp_status = {}')
+	vim.command(
+		'''
+		execute 'function! Ctrlp_status_main(focus, byfname, regex, prev, item, next, marked)
```

Comment:
Vim exceptions are converted into Python exceptions. This cannot possibly be a scope problem because functions in VimL may be defined in exactly one scope: global. There are no other scopes at all, `s:` and anonymous functions included (first are global functions with script-specific prefix (`:h <SID>`), second are global functions with numbers as names).

Note that the error produced by `vim.command` should be caught and redirected to the log file if you have defined the log file (if you have not it will be caught and redirected to `sys.stderr` which means displaying as a message). From two subsequent errors from the same segment with identical error message only first is displayed.